test/Npgsql.Tests/Replication/SafeReplicationTestBase.cs

```diff
@@ -111,10 +111,15 @@ async Task Cleanup()
                     if (pid.Success)
                     {
                         await c.ExecuteNonQueryAsync($"SELECT pg_terminate_backend ({pid.Value})");
+                        for (var i = 0; (bool)(await c.ExecuteScalarAsync($"SELECT EXISTS(SELECT * FROM pg_stat_replication where pid = {pid.Value})"))! && i < 30; i++)
```

Comment:
@Brar since we do know the backend's pid, we just check whether it specifically exists.